### PR TITLE
machine worker label sync

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/33385
https://github.com/rancher/rancher/issues/32654

### CODE REVIEW

File: pkg/controllers/managementuser/machinerole/machinerole.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	machineName := node.Annotations["cluster.x-k8s.io/machine"]
+	if machineName == "" {
+		return node, nil
+	}
+	machineNS := node.Annotations["cluster.x-k8s.io/cluster-namespace"]
+	if machineNS == "" {
+		return node, nil
+	}
+
+	if _, ok := node.Labels["node-role.kubernetes.io/worker"]; ok {
```
Comment (dramich) [inline pkg/controllers/managementuser/machinerole/machinerole.go:42]: This should be the first check

File: pkg/controllers/managementuser/machinerole/machinerole.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		return node, nil
+	}
+	machine, err := h.machines.Get(machineNS, machineName)
+	if err != nil {
+		return nil, err
+	}
+	if val, exists := machine.Labels["rke.cattle.io/worker-role"]; exists && val == "true" {
```
Comment (StrongMonkey) [inline pkg/controllers/managementuser/machinerole/machinerole.go:48]: nit, this should be enough
`machine.Labels["rke.cattle.io/worker-role"] == "true"`

Comment (StrongMonkey) [reply pkg/controllers/managementuser/machinerole/machinerole.go:48]: probably we should update deepcopy only and not modifying `node` directly. Also no need to run `h.nodes.Update(node)` if the above condition is false?

Comment (kinarashah) [reply pkg/controllers/managementuser/machinerole/machinerole.go:48]: thanks, forgot we need deepcopy! updating

**Review by StrongMonkey — APPROVED** [review]

**Review by dramich — APPROVED** [review]

**Pull request merged**